shotover-proxy/src/transforms/protect/mod.rs

```diff
@@ -421,11 +422,38 @@ mod protect_transform_tests {
         panic!()
     }
 
-    //#[tokio::test(flavor = "multi_thread")]
+    /// Creates a new AWS key and returns the key ARN
+    async fn create_aws_key() -> Result<String> {
+        let client = reqwest::Client::new();
+
+        let res = client.post("http://localhost:5000")
+            .body( "{ \"Description\":\"Testing Key\", \
+                               \"KeyUsage\":\"ENCRYPT_DECRYPT\", \
+                               \"CustomerMasterKeySpec\":\"SYMMETRIC_DEFAULT\"}")
+            .header( "Accept-Encoding", "identity" )
+            .header( "Accept", "text/plain")
+            .header( "X-Amz-Target","TrentService.CreateKey")
+            .header( "Content-Type","application/x-amz-json-1.1")
+            .header( "X-Amz-Date","20220126T133630Z")
+            .header( "X-Amz-Security-Token","testing")
+            .header( "Authorization","AWS4-HMAC-SHA256 Credential=testing/20220126/us-east-1/kms/aws4_request, SignedHeaders=content-type;host;x-amz-date;x-amz-security-token;x-amz-target, Signature=7983d06364243f7d9a2a3353d65c724e79bffd7381a10753725a04727a9d8c07")
+            .send().await?;
+
+        let result = res.text().await?;
+        let parsed: serde_json::Value = serde_json::from_str(result.as_str())?;
+        Ok(parsed["KeyMetadata"]["Arn"].as_str().unwrap().to_string())
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
     //#[ignore] // reason: requires AWS credentials
-    #[allow(unused)]
+    //#[allow(unused)]
     // Had to disable this when the repo went public as we dont have access to github secrets anymore
     async fn test_protect_kms_transform() -> Result<()> {
+        let _compose = DockerCompose::new("tests/transforms/docker-compose-moto.yml")
```

Comment:
This is an integration test, so it should go with the other integration tests. You would typically only put unit tests at the bottom of a file.

Reply:
Yeah I wouldnt call this an integration test unless we actually run full shotover against a cassandra instance like our other integration tests do.
Also its already here, moving it in this PR would mess up the diff.

Reply:
Yep, I was incorrect. The `DockerCompose` threw me off.